core/server/common/src/main/java/alluxio/Registry.java

```diff
@@ -36,47 +36,49 @@
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * Class for registering individual masters that make up the Alluxio master service. The intended
- * use of this class is for the Alluxio master service to register the individual masters using an
+ * Class for registering individual servers that run within an Alluxio process. The intended
+ * use of this class is for the Alluxio process to register the individual servers using an
  * instance of this class. The registry is passed as an argument to constructors of individual
- * masters who are expected to add themselves to the registry.
+ * servers who are expected to add themselves to the registry.
  *
  * The reason for using an instance as opposed to a static class is to enable dependency
  * injection in tests.
+ *
+ * @param <T> the type of the server
  */
 @ThreadSafe
-public final class MasterRegistry {
+public final class Registry<T extends Server> {
   private static final int TIMEOUT_SECONDS = 1;
   private static final int RETRY_COUNT = 5;
 
-  private final Map<Class<?>, Master> mRegistry = new HashMap<>();
+  private final Map<Class<?>, T> mRegistry = new HashMap<>();
   private final Lock mLock = new ReentrantLock();
   private final Condition mCondition = mLock.newCondition();
 
   /**
-   * Creates a new instance of {@link MasterRegistry}.
+   * Creates a new instance of {@link Registry}.
    */
-  public MasterRegistry() {}
+  public Registry() {}
 
   /**
-   * Attempts to lookup the master for the given class. Because masters can be looked up and
+   * Attempts to lookup the server for the given class. Because servers can be looked up and
    * added to the registry in parallel, the lookeup is retried several times before giving up.
    *
-   * @param clazz the class of the master to get
-   * @param <T> the type of the master to get
-   * @return the master instance, or null if a type mismatch occurs
+   * @param clazz the class of the server to get
+   * @param <U> the type of the server to get
+   * @return the server instance, or null if a type mismatch occurs
    */
-  public <T> T get(Class<T> clazz) {
-    Master master;
+  public <U> U get(Class<U> clazz) {
```

Comment:
you could add additional type safety by changing the first `<U>` to `<U extends T>`. That way only classes which extend `T` can be passed to the method.